### PR TITLE
Drag an area on the allocations calendar to add a holiday

### PR DESCRIPTION
**Pull request by AndrewVos**
image

### CODE REVIEW

File: app/assets/javascripts/modules/calendars/allocations.es6 [hunk 1]
```diff
@@ -38,19 +41,11 @@
...
-        console.log('Holiday guider name', resource.title);
-
-        // @todo Jquery ajax request to POST a new holiday
-        // in the onsuccess method we will need to call
-        // this.$el.fullCalendar('refetchEvents');
-        // this.showAlert('.alert-success');
-        //
-        // or call this.showAlert('.alert-danger');
-        // on failure
+        this.$holidayForm.find('.js-user-id').val(resource.id);
```
Comment (Guntrisoft) [inline app/assets/javascripts/modules/calendars/allocations.es6]: Any reason why this is not done as a `$.post()` https://api.jquery.com/jquery.post/ ?

Comment (AndrewVos) [inline app/assets/javascripts/modules/calendars/allocations.es6]: Mostly cause I started to get all paranoid about CSRF and thought I should do the safest thing.
I'm not 100% sure that's valid I should probably learn rails a bit more 🤑

File: app/assets/javascripts/modules/calendars/allocations.es6 [hunk 2]
```diff
@@ -16,14 +16,18 @@
...
+        },
+        selectable: true
       };
 
       this.eventChanges = [];
       this.isFullscreen = false;
       this.actionPanel = $('[data-action-panel]');
       this.$savedChanges = $('.js-saved-changes');
       this.$form = $('.js-changes-form');
+      this.$holidayForm = $('.js-holiday-form');
```
Comment (AndrewVos) [inline app/assets/javascripts/modules/calendars/allocations.es6]: Yeah I'm all for that jquery stuff. I hated it in the beginning thinking it was like prefixing the type (`strName`) but it's really really good for readibility!

Comment (Guntrisoft) [inline app/assets/javascripts/modules/calendars/allocations.es6]: yup, totally agree! - would be great if we can enforce it with a linting rule?

File: app/assets/javascripts/modules/calendars/allocations.es6 [hunk 3]
```diff
@@ -33,6 +37,20 @@
       this.setupUndo();
     }
 
+    select(start, end, jsEvent, view, resource) {
+      let title;
+
+      if (title = prompt(`Name of holiday period for ${resource.title}?`)) {
```
Comment (AndrewVos) [inline app/assets/javascripts/modules/calendars/allocations.es6]: @Guntrisoft I can inline the title variable here right?

**Comment by benlovell** [comment]
As discussed, I think `batch_create` makes sense over `create_batch`.

**Pull request merged**